sky/task.py

```diff
@@ -494,6 +505,9 @@ def update_envs(
                     'envs must be List[Tuple[str, str]] or Dict[str, str]: '
                     f'{envs}')
         self._envs.update(envs)
+        if _check_docker_login_config(self._envs):
+            self.resources = _with_docker_login_config(self.resources,
+                                                       self._envs)
```

Comment:
Add a comment for why we are doing this?

Reply:
Done. Thanks!